src/client.rs

```diff
@@ -20,14 +19,17 @@ pub use object::ObjectClient;
 pub use object_access_control::ObjectAccessControlClient;
 
 /// The primary entrypoint to perform operations with Google Cloud Storage.
-pub struct Client {
+pub struct Client<R: RefreshableToken> {
```

Comment:
Do we want to have `R` default to `Token`, so that the upgrade path is as smooth as possible?

Reply:
So after attempting a few thing, I'm not sure what you mean here. An implementation of `Default<Token>` for Client? or a generic trait implementation that defaults to Token